backend/types/events.ts

```diff
@@ -0,0 +1,73 @@
+export interface Event {
+  id: string;
+  name: string;
+  location: string;
+  url?: string;
+  startDate: Date;
+  endDate: Date;
+  source: string;
+  description?: string;
+}
+
+export interface GoogleCalendarItem {
+  kind: "calendar#event";
+  id: string;
+  status: string;
+  htmlLink: string;
+  updated: string;
+  summary: string;
+  location: string;
+  creator: { email: string; displayName: string };
+  organizer: {
+    email: string;
+    displayName: string;
+    self: boolean;
+  };
+  start: { dateTime: string };
+  end: { dateTime: string };
+  transparency: string;
+  iCalUID: string;
+  sequence: number;
+  extendedProperties: { private: { everyoneDeclinedDismissed: string } };
+  guestsCanSeeOtherGuests: boolean;
+}
+
+export interface GoogleCalendarItems {
+  items: GoogleCalendarItem[];
+}
+
+export type EventFetcher = (name: string) => Promise<Event[]>;
+
+export interface EventSource {
+  name: string;
+  fetcher: EventFetcher;
+}
+
+export interface MeetupEvent {
+  id: string;
+  name: string;
+  time: string;
+  duration: number;
+  local_date: string;
+  local_time: string;
+  utc_offset: number;
+  yes_rsvp_count: number;
+  venue: {
+    id: number;
+    name: string;
+    lat: number;
+    long: number;
+    address_1: string;
+    address_2: string;
+    city: string;
+    country: string;
+    zip: string;
+    state: string;
+  };
+  group: {
+    id: number;
+    name: string;
+  };
+  link: string;
+  description: string;
+}
```

Comment:
Not a change request, really... just awareness:
>Since this file is only types, you could change the file extension to `.ts.d` and remove the `export` calls.  This should add the types to the project globally and you wouldn't need to import them into the dependent files.

Reply:
Can you clarify? I tried renaming it to .d.ts and removing the export, but TS wasn't aware of those types in calendar.ts when I removed to import.

Reply:
ts.d ?

Reply:
Sorry.. i'm dumb.   Lemme see

Reply:
Works for me.

You have to remove _all_ `export`s from the file for ts to realize its a definition file

Reply:
I learned something new today, thanks Davin!